src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -371,17 +375,19 @@ public boolean visible(String name, Map<String, Object> connectorConfigs) {
   }
 
   public static ConfigDef getConfig() {
-    Map<String, ConfigDef.ConfigKey> everything = new LinkedHashMap<>(newConfigDef().configKeys());
-    everything.putAll(StorageCommonConfig.newConfigDef(STORAGE_CLASS_RECOMMENDER).configKeys());
-    everything.putAll(PartitionerConfig.newConfigDef(PARTITIONER_CLASS_RECOMMENDER).configKeys());
-    everything.putAll((HiveConfig.getConfig().configKeys()));
-
+    // Define the names of the configurations we're going to override
     Set<String> skip = new HashSet<>();
     skip.add(STORAGE_CLASS_CONFIG);
     skip.add(FORMAT_CLASS_CONFIG);
 
+    // Order added is important, so that group order is maintained
     ConfigDef visible = new ConfigDef();
+    addAllConfigKeys(visible, newConfigDef(), skip);
+    addAllConfigKeys(visible, StorageCommonConfig.newConfigDef(STORAGE_CLASS_RECOMMENDER), skip);
+    addAllConfigKeys(visible, PartitionerConfig.newConfigDef(STORAGE_CLASS_RECOMMENDER), skip);
+    addAllConfigKeys(visible, HiveConfig.getConfig(), skip);
 
+    // Add the overridden configurations
```

Comment:
@kkonstantine this is the same approach I used in https://github.com/confluentinc/kafka-connect-storage-cloud/pull/118: group ordering is actually insertion order in ConfigDef, so we have to add the groups in the same order. We could use a LinkedHashMap, but we'd have to add the overridden configkeys last, and if we're doing that its simpler to just create the ConfigDef rather than a temp LinkedHashMap.